### ISSUE TITLE
Error 500 when creating PKI role from CLI

### ISSUE THREAD

**Issue by kgutwin** [hashicorp/vault#381]
See PR #310 and ping @jefferai. When following the instructions at https://github.com/hashicorp/vault/blob/master/website/source/docs/secrets/pki/index.html.md, I can't write to `pki/roles/example-dot-com`.

```
$ vault mount pki
Successfully mounted 'pki' at 'pki!'
$ vault write pki/config/ca pem_bundle="@ca-bundle.pem"
Success! Data written to: pki/config/ca
$ vault write pki/roles/example-dot-com allowed_base_domain="example.com" allow_subdomains="true" lease_max="72h"
Error writing data to pki/roles/example-dot-com: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/pki/roles/example-dot-com
Code: 500. Errors:

* unsupported path
```

The API documentation only references POST, GET or DELETE operations on `/pki/roles/`, so perhaps PUT is not implemented?

**Comment by jefferai** [comment]
Interesting -- it appears that Vault is unhappy with paths containing hyphens:
```
$ vault write pki/roles/example-dot-com allowed_base_domain="example.com" allow_subdomains="true" lease_max="72h"
Error writing data to pki/roles/example-dot-com: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/pki/roles/example-dot-com
Code: 500. Errors:

* unsupported path

$ vault write pki/roles/exampledotcom allowed_base_domain="example.com" allow_subdomains="true" lease_max="72h"
Success! Data written to: pki/roles/exampledotcom
```
@armon Is this intended, and I should fix the documentation, or a bug?

**Comment by armon** [comment]
@jefferai The regex in the `pki` backend does not allow a hyphen in the path. Vault has no issues with it, but the regex is overly restricted

**Comment by jefferai** [comment]
@armon Oh shucks, you're right. I copied that regex from the Postgres backend and forgot about it, so assumed it was path checking within Vault before hitting the backend.

Pull request with the fix in #386.

**Comment by kgutwin** [comment]
Verified, thanks!
